### PR TITLE
[agent] Update kubernetes_state_core documentation

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

* Document the tags deprecated in `kubernetes_state` check and removed in `kubernetes_state_core`.
* Explain how to configure the `kubernetes_state_core` with the operator.

### Motivation

ease the migration to `kubernetes_state_core`

### Preview

https://docs-staging.datadoghq.com/clamoriniere/ksm-core-migration/integrations/kubernetes_state_core

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/integrations/kubernetes_state_core.md [hunk 1]
```diff
@@ -58,11 +58,62 @@ datadog:
...
 
+### Tags removal
+
+With `kubernetes_state`, several tags were flagged as deprecated and replaced by new tags. To provide a migration path check both tags was send with the metrics.
+
+With `kubernetes_state_core` only the none deprecated tags are send.
+
+Before migrating to `kubernetes_state_core`, verify that only official tags are used in monitors and dashboards.
+
+Here is the mapping between deprecated and official tags
```
Comment (jtappa) [inline content/en/integrations/kubernetes_state_core.md:98]:
```suggestion
Here is the mapping between deprecated tags and the official tags that have replaced them:
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Comment by clamoriniere** [comment]
Hi @jtappa 
Thank you for the fast review :)
I have added your comment to the PR.

File: content/en/integrations/kubernetes_state_core.md [hunk 2]
```diff
@@ -58,11 +58,62 @@ datadog:
 ...
 ```
 
+{{% /tab %}}
+{{% tab "Operator" %}}
+
+From the Datadog-Operator v0.7.0
```
Comment (celenechang) [inline content/en/integrations/kubernetes_state_core.md:64]: Maybe we should have this as a note, e.g. `Note: Datadog Operator v.0.7.0 or greater is required.`

File: content/en/integrations/kubernetes_state_core.md [hunk 3]
```diff
@@ -58,11 +58,62 @@ datadog:
...
+```
+
 {{% /tab %}}
 {{< /tabs >}}
 
 ## Migration from kubernetes_state to kubernetes_state_core
 
+### Tags removal
+
+With `kubernetes_state`, several tags have been flagged as deprecated and replaced by new tags. To determine your migration path, check which tags are sent with your metrics.
```
Comment (celenechang) [inline content/en/integrations/kubernetes_state_core.md:92]:
```suggestion
In the original `kubernetes_state` check, several tags have been flagged as deprecated and replaced by new tags. To determine your migration path, check which tags are submitted with your metrics.
```

File: content/en/integrations/kubernetes_state_core.md [hunk 4]
```diff
@@ -58,11 +58,62 @@ datadog:
...
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
+    appKey: <DATADOG_APP_KEY>
+  features:
+    kubeStateMetricsCore:
+      enabled: true
+  # ...
+```
+
```
Comment (clamoriniere) [inline content/en/integrations/kubernetes_state_core.md:84]:
```suggestion

Note: Datadog Operator v.0.7.0 or greater is required.

```

File: content/en/integrations/kubernetes_state_core.md [hunk 5]
```diff
@@ -58,11 +58,62 @@ datadog:
...
 
 ## Migration from kubernetes_state to kubernetes_state_core
 
+### Tags removal
+
+With `kubernetes_state`, several tags have been flagged as deprecated and replaced by new tags. To determine your migration path, check which tags are sent with your metrics.
+
+With `kubernetes_state_core`, only the non-deprecated tags are sent.
+
+Before migrating to `kubernetes_state_core`, verify that only official tags are used in monitors and dashboards.
```
Comment (celenechang) [inline content/en/integrations/kubernetes_state_core.md:96]:
```suggestion
In the `kubernetes_state_core` check, only the non-deprecated tags are submitted. Before migrating from `kubernetes_state` to `kubernetes_state_core`, verify that only official tags are used in monitors and dashboards.
```

File: content/en/integrations/kubernetes_state_core.md [hunk 6]
```diff
@@ -58,11 +58,62 @@ datadog:
 ...
 ```
 
+{{% /tab %}}
+{{% tab "Operator" %}}
+
+From the Datadog-Operator v0.7.0
+
```
Comment (clamoriniere) [inline content/en/integrations/kubernetes_state_core.md:65]:
```suggestion
```

File: content/en/integrations/kubernetes_state_core.md (hunk 2 above)
Comment (clamoriniere) [reply content/en/integrations/kubernetes_state_core.md:64]: I move it at the bottom of this tab since it is a note

**Review by jtappa — APPROVED** [review]

**Comment by jtappa** [comment]
@clamoriniere Is this okay to merge, or is it waiting on a release?

**Comment by clamoriniere** [comment]
Ok to merge

**Pull request merged**